src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -422,17 +422,14 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
     fun addAnnotation(annotation: KClass<*>) = addAnnotation(annotation.asClassName())
 
     fun addModifiers(vararg modifiers: KModifier) = apply {
-      check(anonymousTypeArguments == null) { "forbidden on anonymous types." }
```

Comment:
Is there anything replacing these checks? I'm fine with just removing them, but it'd be nice to keep them

Reply:
yes, i think we can do the same `isAnonymousClass` check from above. good idea!

Reply:
Would be nice to have tests for this in the future

Reply:
#335